diff --git a/dist/index.js b/dist/index.js
index ec7e5f9b3436a099cedd9362eea7a4cf8a9f5bfd..c8762a81af718a5a3e4b24c1ce036973bfef04f4 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -9,7 +9,9 @@ import * as chai from 'chai';
 export { chai };
 
 const MATCHERS_OBJECT = Symbol.for("matchers-object");
-const JEST_MATCHERS_OBJECT = Symbol.for("$$jest-matchers-object");
+// Patched this symbol for storybook, so that storybook/test can be used in a jest environment as well.
+// Otherwise, vitest will override global jest matchers, and crash.
+const JEST_MATCHERS_OBJECT = Symbol.for("$$jest-matchers-object-storybook");
 const GLOBAL_EXPECT = Symbol.for("expect-global");
 const ASYMMETRIC_MATCHERS_OBJECT = Symbol.for("asymmetric-matchers-object");
 
@@ -671,20 +673,24 @@ function isStandardSchema(obj) {
 	return !!obj && (typeof obj === "object" || typeof obj === "function") && obj["~standard"] && typeof obj["~standard"].validate === "function";
 }
 
-if (!Object.hasOwn(globalThis, MATCHERS_OBJECT)) {
+if (!Object.prototype.hasOwnProperty.call(globalThis, MATCHERS_OBJECT)) {
 	const globalState = new WeakMap();
-	const matchers = Object.create(null);
-	const customEqualityTesters = [];
-	const asymmetricMatchers = Object.create(null);
 	Object.defineProperty(globalThis, MATCHERS_OBJECT, { get: () => globalState });
+}
+if (!Object.prototype.hasOwnProperty.call(globalThis, JEST_MATCHERS_OBJECT)) {
+  const matchers = Object.create(null);
+  const customEqualityTesters = [];
 	Object.defineProperty(globalThis, JEST_MATCHERS_OBJECT, {
 		configurable: true,
 		get: () => ({
-			state: globalState.get(globalThis[GLOBAL_EXPECT]),
+			state: globalThis[MATCHERS_OBJECT].get(globalThis[GLOBAL_EXPECT]),
 			matchers,
 			customEqualityTesters
 		})
 	});
+}
+if (!Object.prototype.hasOwnProperty.call(globalThis, ASYMMETRIC_MATCHERS_OBJECT)) {
+  const asymmetricMatchers = Object.create(null);
 	Object.defineProperty(globalThis, ASYMMETRIC_MATCHERS_OBJECT, { get: () => asymmetricMatchers });
 }
 function getState(expect) {
