diff --git a/dist/Button.mjs b/dist/Button.mjs
index ee2311efa2297f8478d2258d3468a614e9a3d3f4..cefaff17718b2027514e97bfdc96a453ac865bc1 100644
--- a/dist/Button.mjs
+++ b/dist/Button.mjs
@@ -1,9 +1,11 @@
 import {dom as $64fa3d84918910a7$export$df3a06d6289f983e, useContextProps as $64fa3d84918910a7$export$29f1550f4b0d4415, useRenderProps as $64fa3d84918910a7$export$4d86445c2cf5e3} from "./utils.mjs";
 import {ProgressBarContext as $0393f8ab869a0f1a$export$e9f3bf65a26ce129} from "./ProgressBar.mjs";
 import {announce as $fM325$announce} from "@react-aria/live-announcer";
-import {useButton as $fM325$useButton, useFocusRing as $fM325$useFocusRing, useHover as $fM325$useHover, useId as $fM325$useId, mergeProps as $fM325$mergeProps} from "react-aria";
+import {useButton as $fM325$useButton} from "@react-aria/button";
+import {useFocusRing as $fM325$useFocusRing} from "@react-aria/focus";
+import {useHover as $fM325$useHover} from "@react-aria/interactions";
+import {useId as $fM325$useId, mergeProps as $fM325$mergeProps, filterDOMProps as $fM325$filterDOMProps} from "@react-aria/utils";
 import {createHideableComponent as $fM325$createHideableComponent} from "@react-aria/collections";
-import {filterDOMProps as $fM325$filterDOMProps} from "@react-aria/utils";
 import $fM325$react, {createContext as $fM325$createContext, useRef as $fM325$useRef, useEffect as $fM325$useEffect} from "react";
 
 /*
diff --git a/dist/Dialog.mjs b/dist/Dialog.mjs
index 52ff01117b1734651b0dfedae9efa0589ed003c5..f384212ce565e045d5cb56068525989616ade1d0 100644
--- a/dist/Dialog.mjs
+++ b/dist/Dialog.mjs
@@ -3,9 +3,11 @@ import {DEFAULT_SLOT as $64fa3d84918910a7$export$c62b8e45d58ddad9, dom as $64fa3
 import {HeadingContext as $4e85f108e88277b8$export$d688439359537581} from "./RSPContexts.mjs";
 import {PopoverContext as $07b14b47974efb58$export$9b9a0cd73afb7ca4} from "./Popover.mjs";
 import {RootMenuTriggerStateContext as $3674c52c6b3c5bce$export$795aec4671cbae19} from "./Menu.mjs";
-import {useOverlayTrigger as $6IYYA$useOverlayTrigger, useId as $6IYYA$useId, useDialog as $6IYYA$useDialog} from "react-aria";
+import {useOverlayTrigger as $6IYYA$useOverlayTrigger} from "@react-aria/overlays";
+import {useDialog as $6IYYA$useDialog} from "@react-aria/dialog";
+import {useId as $6IYYA$useId} from "@react-aria/utils";
 import {useResizeObserver as $6IYYA$useResizeObserver, filterDOMProps as $6IYYA$filterDOMProps, mergeProps as $6IYYA$mergeProps} from "@react-aria/utils";
-import {useMenuTriggerState as $6IYYA$useMenuTriggerState} from "react-stately";
+import {useMenuTriggerState as $6IYYA$useMenuTriggerState} from "@react-stately/menu";
 import {PressResponder as $6IYYA$PressResponder} from "@react-aria/interactions";
 import $6IYYA$react, {createContext as $6IYYA$createContext, useRef as $6IYYA$useRef, useState as $6IYYA$useState, useCallback as $6IYYA$useCallback, forwardRef as $6IYYA$forwardRef, useContext as $6IYYA$useContext} from "react";
 
diff --git a/dist/Menu.mjs b/dist/Menu.mjs
index b386cc2b514562cc67ef0aac236dd420ee1a06bf..8150f6dca4ef8f9aaee4d2284ccdb1ced6d2b9c7 100644
--- a/dist/Menu.mjs
+++ b/dist/Menu.mjs
@@ -9,10 +9,14 @@ import {SelectionIndicatorContext as $1d5b8b8664671ef2$export$c9549807523555e0}
 import {SeparatorContext as $431f98aba6844401$export$6615d83f6de245ce} from "./Separator.mjs";
 import {SharedElementTransition as $c8a5a149f625efcf$export$758399f318e6385a} from "./SharedElementTransition.mjs";
 import {TextContext as $514c0188e459b4c0$export$9afb8bc826b033ea} from "./Text.mjs";
-import {useMenuTrigger as $kM2ZM$useMenuTrigger, useSubmenuTrigger as $kM2ZM$useSubmenuTrigger, useMenu as $kM2ZM$useMenu, FocusScope as $kM2ZM$FocusScope, mergeProps as $kM2ZM$mergeProps, useMenuSection as $kM2ZM$useMenuSection, useMenuItem as $kM2ZM$useMenuItem, useHover as $kM2ZM$useHover} from "react-aria";
+import {useMenuTrigger as $kM2ZM$useMenuTrigger, useSubmenuTrigger as $kM2ZM$useSubmenuTrigger} from "@react-aria/menu";
+import {useMenu as $kM2ZM$useMenu, useMenuSection as $kM2ZM$useMenuSection, useMenuItem as $kM2ZM$useMenuItem} from "@react-aria/menu";
+import {FocusScope as $kM2ZM$FocusScope} from "@react-aria/focus";
+import {mergeProps as $kM2ZM$mergeProps, useResizeObserver as $kM2ZM$useResizeObserver, useObjectRef as $kM2ZM$useObjectRef, filterDOMProps as $kM2ZM$filterDOMProps} from "@react-aria/utils";
+import {useHover as $kM2ZM$useHover} from "@react-aria/interactions";
 import {CollectionNode as $kM2ZM$CollectionNode, createBranchComponent as $kM2ZM$createBranchComponent, CollectionBuilder as $kM2ZM$CollectionBuilder, Collection as $kM2ZM$Collection, SectionNode as $kM2ZM$SectionNode, createLeafComponent as $kM2ZM$createLeafComponent, ItemNode as $kM2ZM$ItemNode} from "@react-aria/collections";
-import {useMenuTriggerState as $kM2ZM$useMenuTriggerState, useSubmenuTriggerState as $kM2ZM$useSubmenuTriggerState, useTreeState as $kM2ZM$useTreeState} from "react-stately";
-import {useResizeObserver as $kM2ZM$useResizeObserver, useObjectRef as $kM2ZM$useObjectRef, filterDOMProps as $kM2ZM$filterDOMProps} from "@react-aria/utils";
+import {useMenuTriggerState as $kM2ZM$useMenuTriggerState, useSubmenuTriggerState as $kM2ZM$useSubmenuTriggerState} from "@react-stately/menu";
+import {useTreeState as $kM2ZM$useTreeState} from "@react-stately/tree";
 import {SelectionManager as $kM2ZM$SelectionManager, useMultipleSelectionState as $kM2ZM$useMultipleSelectionState} from "@react-stately/selection";
 import {PressResponder as $kM2ZM$PressResponder} from "@react-aria/interactions";
 import $kM2ZM$react, {createContext as $kM2ZM$createContext, useRef as $kM2ZM$useRef, useState as $kM2ZM$useState, useCallback as $kM2ZM$useCallback, useContext as $kM2ZM$useContext, forwardRef as $kM2ZM$forwardRef, useMemo as $kM2ZM$useMemo} from "react";
diff --git a/dist/Modal.mjs b/dist/Modal.mjs
index 357186db64e26e5bcc1a3f02423ddadfe147919c..06264c4a777fe4c4f289120a5b1ac04cb7330818 100644
--- a/dist/Modal.mjs
+++ b/dist/Modal.mjs
@@ -1,8 +1,9 @@
 import {dom as $64fa3d84918910a7$export$df3a06d6289f983e, Provider as $64fa3d84918910a7$export$2881499e37b75b9a, useContextProps as $64fa3d84918910a7$export$29f1550f4b0d4415, useRenderProps as $64fa3d84918910a7$export$4d86445c2cf5e3} from "./utils.mjs";
 import {OverlayTriggerStateContext as $de32f1b87079253c$export$d2f961adcb0afbe} from "./Dialog.mjs";
-import {useIsSSR as $daTMi$useIsSSR, useModalOverlay as $daTMi$useModalOverlay, Overlay as $daTMi$Overlay, DismissButton as $daTMi$DismissButton} from "react-aria";
+import {useIsSSR as $daTMi$useIsSSR} from "@react-aria/ssr";
+import {useModalOverlay as $daTMi$useModalOverlay, Overlay as $daTMi$Overlay, DismissButton as $daTMi$DismissButton} from "@react-aria/overlays";
 import {useObjectRef as $daTMi$useObjectRef, useExitAnimation as $daTMi$useExitAnimation, useEnterAnimation as $daTMi$useEnterAnimation, useViewportSize as $daTMi$useViewportSize, isScrollable as $daTMi$isScrollable, mergeProps as $daTMi$mergeProps, filterDOMProps as $daTMi$filterDOMProps, mergeRefs as $daTMi$mergeRefs} from "@react-aria/utils";
-import {useOverlayTriggerState as $daTMi$useOverlayTriggerState} from "react-stately";
+import {useOverlayTriggerState as $daTMi$useOverlayTriggerState} from "@react-stately/overlays";
 import $daTMi$react, {createContext as $daTMi$createContext, forwardRef as $daTMi$forwardRef, useContext as $daTMi$useContext, useRef as $daTMi$useRef, useMemo as $daTMi$useMemo} from "react";
 
 /*
diff --git a/dist/Popover.mjs b/dist/Popover.mjs
index 9df29d4b4a5ceb8a68ddd0501b5314454fce1e45..2c10dbf583ebe86e0ff22af4e1990fbb98a96283 100644
--- a/dist/Popover.mjs
+++ b/dist/Popover.mjs
@@ -1,10 +1,11 @@
 import {dom as $64fa3d84918910a7$export$df3a06d6289f983e, useContextProps as $64fa3d84918910a7$export$29f1550f4b0d4415, useRenderProps as $64fa3d84918910a7$export$4d86445c2cf5e3} from "./utils.mjs";
 import {OverlayArrowContext as $44f671af83e7d9e0$export$2de4954e8ae13b9f} from "./OverlayArrow.mjs";
 import {OverlayTriggerStateContext as $de32f1b87079253c$export$d2f961adcb0afbe} from "./Dialog.mjs";
-import {useLocale as $ehFet$useLocale, usePopover as $ehFet$usePopover, DismissButton as $ehFet$DismissButton, Overlay as $ehFet$Overlay} from "react-aria";
+import {useLocale as $ehFet$useLocale} from "@react-aria/i18n";
+import {usePopover as $ehFet$usePopover, DismissButton as $ehFet$DismissButton, Overlay as $ehFet$Overlay} from "@react-aria/overlays";
 import {useExitAnimation as $ehFet$useExitAnimation, useEnterAnimation as $ehFet$useEnterAnimation, useLayoutEffect as $ehFet$useLayoutEffect, isFocusWithin as $ehFet$isFocusWithin, mergeProps as $ehFet$mergeProps, filterDOMProps as $ehFet$filterDOMProps} from "@react-aria/utils";
 import {getInteractionModality as $ehFet$getInteractionModality, focusSafely as $ehFet$focusSafely} from "@react-aria/interactions";
-import {useOverlayTriggerState as $ehFet$useOverlayTriggerState} from "react-stately";
+import {useOverlayTriggerState as $ehFet$useOverlayTriggerState} from "@react-stately/overlays";
 import $ehFet$react, {createContext as $ehFet$createContext, forwardRef as $ehFet$forwardRef, useContext as $ehFet$useContext, useRef as $ehFet$useRef, useState as $ehFet$useState, useEffect as $ehFet$useEffect, useMemo as $ehFet$useMemo} from "react";
 import {useIsHidden as $ehFet$useIsHidden} from "@react-aria/collections";
 
diff --git a/dist/ProgressBar.mjs b/dist/ProgressBar.mjs
index 7662d6460e2ffcad54f7065750244892b3af376c..257703a026cd94019369122fa97b8de8886ed806 100644
--- a/dist/ProgressBar.mjs
+++ b/dist/ProgressBar.mjs
@@ -1,6 +1,6 @@
 import {dom as $64fa3d84918910a7$export$df3a06d6289f983e, useContextProps as $64fa3d84918910a7$export$29f1550f4b0d4415, useRenderProps as $64fa3d84918910a7$export$4d86445c2cf5e3, useSlot as $64fa3d84918910a7$export$9d4c57ee4c6ffdd8} from "./utils.mjs";
 import {LabelContext as $01b77f81d0f07f68$export$75b6ee27786ba447} from "./Label.mjs";
-import {useProgressBar as $hU2kz$useProgressBar} from "react-aria";
+import {useProgressBar as $hU2kz$useProgressBar} from "@react-aria/progress";
 import {clamp as $hU2kz$clamp} from "@react-stately/utils";
 import {filterDOMProps as $hU2kz$filterDOMProps, mergeProps as $hU2kz$mergeProps} from "@react-aria/utils";
 import $hU2kz$react, {createContext as $hU2kz$createContext, forwardRef as $hU2kz$forwardRef} from "react";
diff --git a/dist/Separator.mjs b/dist/Separator.mjs
index f1ddc4ef8883d875d3fa7888e34132ba838d8fb4..7e643d9b25a1d618d1152caf35740dbb5b1b9c75 100644
--- a/dist/Separator.mjs
+++ b/dist/Separator.mjs
@@ -1,5 +1,5 @@
 import {dom as $64fa3d84918910a7$export$df3a06d6289f983e, useContextProps as $64fa3d84918910a7$export$29f1550f4b0d4415} from "./utils.mjs";
-import {useSeparator as $i9JCE$useSeparator} from "react-aria";
+import {useSeparator as $i9JCE$useSeparator} from "@react-aria/separator";
 import {CollectionNode as $i9JCE$CollectionNode, createLeafComponent as $i9JCE$createLeafComponent} from "@react-aria/collections";
 import {filterDOMProps as $i9JCE$filterDOMProps, mergeProps as $i9JCE$mergeProps} from "@react-aria/utils";
 import $i9JCE$react, {createContext as $i9JCE$createContext} from "react";
diff --git a/dist/Tabs.mjs b/dist/Tabs.mjs
index 22dc6db50ec75ee3d7a2167334f49882a08c4353..8704089150f22ce4b4480090f4aa5b070a092942 100644
--- a/dist/Tabs.mjs
+++ b/dist/Tabs.mjs
@@ -2,10 +2,13 @@ import {dom as $64fa3d84918910a7$export$df3a06d6289f983e, Provider as $64fa3d849
 import {CollectionRendererContext as $7135fc7d473fd974$export$4feb769f8ddf26c5, DefaultCollectionRenderer as $7135fc7d473fd974$export$a164736487e3f0ae, usePersistedKeys as $7135fc7d473fd974$export$90e00781bc59d8f9} from "./Collection.mjs";
 import {SelectionIndicatorContext as $1d5b8b8664671ef2$export$c9549807523555e0} from "./SelectionIndicator.mjs";
 import {SharedElementTransition as $c8a5a149f625efcf$export$758399f318e6385a} from "./SharedElementTransition.mjs";
-import {useFocusRing as $7aSLZ$useFocusRing, mergeProps as $7aSLZ$mergeProps, useTabList as $7aSLZ$useTabList, useTab as $7aSLZ$useTab, useHover as $7aSLZ$useHover, useTabPanel as $7aSLZ$useTabPanel} from "react-aria";
+import {useFocusRing as $7aSLZ$useFocusRing} from "@react-aria/focus";
+import {mergeProps as $7aSLZ$mergeProps} from "@react-aria/utils";
+import {useTabList as $7aSLZ$useTabList, useTab as $7aSLZ$useTab, useTabPanel as $7aSLZ$useTabPanel} from "@react-aria/tabs";
+import {useHover as $7aSLZ$useHover} from "@react-aria/interactions";
 import {CollectionBuilder as $7aSLZ$CollectionBuilder, Collection as $7aSLZ$Collection, CollectionNode as $7aSLZ$CollectionNode, createLeafComponent as $7aSLZ$createLeafComponent, createHideableComponent as $7aSLZ$createHideableComponent} from "@react-aria/collections";
 import {filterDOMProps as $7aSLZ$filterDOMProps, useObjectRef as $7aSLZ$useObjectRef, useLayoutEffect as $7aSLZ$useLayoutEffect, useExitAnimation as $7aSLZ$useExitAnimation, useEnterAnimation as $7aSLZ$useEnterAnimation, inertValue as $7aSLZ$inertValue} from "@react-aria/utils";
-import {useTabListState as $7aSLZ$useTabListState} from "react-stately";
+import {useTabListState as $7aSLZ$useTabListState} from "@react-stately/tabs";
 import $7aSLZ$react, {createContext as $7aSLZ$createContext, forwardRef as $7aSLZ$forwardRef, useMemo as $7aSLZ$useMemo, useContext as $7aSLZ$useContext, useRef as $7aSLZ$useRef, useState as $7aSLZ$useState} from "react";
 
 /*
diff --git a/dist/Tooltip.mjs b/dist/Tooltip.mjs
index eb50cd757a2744d487dca3042cb3da668bd44441..ee4885697bb5b4a8754e4bf9a440f6294250811b 100644
--- a/dist/Tooltip.mjs
+++ b/dist/Tooltip.mjs
@@ -1,9 +1,11 @@
 import {dom as $64fa3d84918910a7$export$df3a06d6289f983e, Provider as $64fa3d84918910a7$export$2881499e37b75b9a, useContextProps as $64fa3d84918910a7$export$29f1550f4b0d4415, useRenderProps as $64fa3d84918910a7$export$4d86445c2cf5e3} from "./utils.mjs";
 import {OverlayArrowContext as $44f671af83e7d9e0$export$2de4954e8ae13b9f} from "./OverlayArrow.mjs";
-import {useTooltipTrigger as $cCslV$useTooltipTrigger, OverlayContainer as $cCslV$OverlayContainer, useOverlayPosition as $cCslV$useOverlayPosition, mergeProps as $cCslV$mergeProps, useTooltip as $cCslV$useTooltip} from "react-aria";
-import {useExitAnimation as $cCslV$useExitAnimation, useEnterAnimation as $cCslV$useEnterAnimation, filterDOMProps as $cCslV$filterDOMProps} from "@react-aria/utils";
+import {useTooltipTrigger as $cCslV$useTooltipTrigger} from "@react-aria/tooltip";
+import {OverlayContainer as $cCslV$OverlayContainer, useOverlayPosition as $cCslV$useOverlayPosition} from "@react-aria/overlays";
+import {mergeProps as $cCslV$mergeProps, useExitAnimation as $cCslV$useExitAnimation, useEnterAnimation as $cCslV$useEnterAnimation, filterDOMProps as $cCslV$filterDOMProps} from "@react-aria/utils";
+import {useTooltip as $cCslV$useTooltip} from "@react-aria/tooltip";
 import {FocusableProvider as $cCslV$FocusableProvider} from "@react-aria/focus";
-import {useTooltipTriggerState as $cCslV$useTooltipTriggerState} from "react-stately";
+import {useTooltipTriggerState as $cCslV$useTooltipTriggerState} from "@react-stately/tooltip";
 import $cCslV$react, {createContext as $cCslV$createContext, useRef as $cCslV$useRef, forwardRef as $cCslV$forwardRef, useContext as $cCslV$useContext} from "react";
 
 /*
diff --git a/dist/Tree.mjs b/dist/Tree.mjs
index 605f99eb7313afa218ed605e2948bf18067dc494..e2b5465b0f5e7a4e197ed5a3ce94e71a8b08f751 100644
--- a/dist/Tree.mjs
+++ b/dist/Tree.mjs
@@ -7,9 +7,16 @@ import {GridListHeader as $72e60046c03fbe42$export$1b574dbdb0075ff6, GridListHea
 import {SelectionIndicatorContext as $1d5b8b8664671ef2$export$c9549807523555e0} from "./SelectionIndicator.mjs";
 import {SharedElementTransition as $c8a5a149f625efcf$export$758399f318e6385a} from "./SharedElementTransition.mjs";
 import {TreeDropTargetDelegate as $7e0f784ed4e400df$export$82c13862611c034e} from "./TreeDropTargetDelegate.mjs";
-import {useLocale as $kUtXx$useLocale, useCollator as $kUtXx$useCollator, useTree as $kUtXx$useTree, ListKeyboardDelegate as $kUtXx$ListKeyboardDelegate, useFocusRing as $kUtXx$useFocusRing, FocusScope as $kUtXx$FocusScope, mergeProps as $kUtXx$mergeProps, useTreeItem as $kUtXx$useTreeItem, useHover as $kUtXx$useHover, useGridListSelectionCheckbox as $kUtXx$useGridListSelectionCheckbox, useVisuallyHidden as $kUtXx$useVisuallyHidden, useId as $kUtXx$useId, useGridListSection as $kUtXx$useGridListSection} from "react-aria";
+import {useLocale as $kUtXx$useLocale, useCollator as $kUtXx$useCollator} from "@react-aria/i18n";
+import {useTree as $kUtXx$useTree, useTreeItem as $kUtXx$useTreeItem} from "@react-aria/tree";
+import {ListKeyboardDelegate as $kUtXx$ListKeyboardDelegate} from "@react-aria/selection";
+import {useFocusRing as $kUtXx$useFocusRing, FocusScope as $kUtXx$FocusScope} from "@react-aria/focus";
+import {mergeProps as $kUtXx$mergeProps, useId as $kUtXx$useId} from "@react-aria/utils";
+import {useHover as $kUtXx$useHover} from "@react-aria/interactions";
+import {useGridListSelectionCheckbox as $kUtXx$useGridListSelectionCheckbox, useGridListSection as $kUtXx$useGridListSection} from "@react-aria/gridlist";
+import {useVisuallyHidden as $kUtXx$useVisuallyHidden} from "@react-aria/visually-hidden";
 import {BaseCollection as $kUtXx$BaseCollection, CollectionBuilder as $kUtXx$CollectionBuilder, Collection as $kUtXx$Collection, CollectionNode as $kUtXx$CollectionNode, createLeafComponent as $kUtXx$createLeafComponent, createBranchComponent as $kUtXx$createBranchComponent, useCachedChildren as $kUtXx$useCachedChildren, LoaderNode as $kUtXx$LoaderNode, SectionNode as $kUtXx$SectionNode} from "@react-aria/collections";
-import {useTreeState as $kUtXx$useTreeState} from "react-stately";
+import {useTreeState as $kUtXx$useTreeState} from "@react-stately/tree";
 import {filterDOMProps as $kUtXx$filterDOMProps, useObjectRef as $kUtXx$useObjectRef, useLoadMoreSentinel as $kUtXx$useLoadMoreSentinel, inertValue as $kUtXx$inertValue} from "@react-aria/utils";
 import $kUtXx$react, {createContext as $kUtXx$createContext, forwardRef as $kUtXx$forwardRef, useRef as $kUtXx$useRef, useEffect as $kUtXx$useEffect, useContext as $kUtXx$useContext, useState as $kUtXx$useState, useMemo as $kUtXx$useMemo} from "react";
 import {useControlledState as $kUtXx$useControlledState} from "@react-stately/utils";
diff --git a/package.json b/package.json
index e9543785443ce6ac2a9e124fc18dbba9decd635f..f9426815865f50058c566ea191eb64811a219e42 100644
--- a/package.json
+++ b/package.json
@@ -26,6 +26,10 @@
       "types": "./i18n/lang.d.ts",
       "import": "./i18n/*.mjs",
       "require": "./i18n/*.js"
+    },
+    "./patched-dist/*": {
+      "import": "./dist/*.mjs",
+      "types": "./dist/types.d.ts"
     }
   },
   "files": [
